TYPO3.Media/Classes/TYPO3/Media/Command/MediaCommandController.php

```diff
@@ -109,13 +124,50 @@ public function importResourcesCommand($simulate = false)
     }
 
     /**
+     * Generate thumbnails for thumbnail presets
+     *
+     * @param string $preset Preset name, if not provided thumbnails are created for all presets
+     * @param boolean $async Asynchronous generation, if not provided the setting ``TYPO3.Media.asyncThumbnails`` is used
+     * @return void
+     */
+    public function createThumbnailsCommand($preset = null, $async = null)
+    {
+        $async = $async !== null ? $async : $this->asyncThumbnails;
+        $presets = $preset !== null ? [$preset] : array_keys($this->thumbnailService->getPresets());
+        $presetThumbnailConfigurations = [];
+        foreach ($presets as $preset) {
+            $presetThumbnailConfigurations[] = $this->thumbnailService->getThumbnailConfigurationForPreset($preset, $async);
+        }
+        $iterator = $this->assetRepository->findAllIterator();
+        $imageCount = $this->assetRepository->countAll();
+        $this->output->progressStart($imageCount * count($presetThumbnailConfigurations));
+        foreach ($this->assetRepository->iterate($iterator) as $image) {
+            foreach ($presetThumbnailConfigurations as $presetThumbnailConfiguration) {
+                $this->thumbnailService->getThumbnail($image, $presetThumbnailConfiguration);
+                $this->persistenceManager->persistAll();
+                $this->output->progressAdvance(1);
+            }
+        }
+    }
+
+    /**
      * Remove all thumbnail objects and resources
+     *
```

Comment:
Also add a help message here, and try to keep the first line shorter ("Remove all thumbnails" would suffice, I guess).